csi/node_server.go

```diff
@@ -118,6 +118,38 @@ func (ns *NodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 		return &csi.NodePublishVolumeResponse{}, nil
 	}
 
+	// we validate the staging path to make sure the global mount is still valid
+	if isMnt, err := ensureMountPoint(stagingPath, mounter); err != nil || !isMnt {
```

Comment:
Quick question, except for RWO encrypted volumes, does the checks (including `ioutil.ReadDir(targetPath)`) in this function work for Block volumes? If YES, then this LGTM.

Reply:
It seems like Block volume is handled by the above code not here, so it should not be impacted by this change?

Reply:
Correct block volumes for the moment aren't handled by NodeStage phase and only handled by NodePublish phase. Since there are some inconsistencies in the documentation regardion NodeStage before for block volumes, I decided to skip implementation in NodeStage since anyway for block volumes we only create a mknod for the associated (major:minor)